vmm/src/memory_manager.rs

```diff
@@ -1242,30 +1233,6 @@ impl MemoryManager {
         Err(Error::UnknownMemoryZone)
     }
 
-    pub fn balloon_resize(&mut self, expected_ram: u64) -> Result<u64, Error> {
-        let mut balloon_size = 0;
-        if let Some(balloon) = &self.balloon {
-            if expected_ram < self.current_ram {
-                balloon_size = self.current_ram - expected_ram;
-            }
-            balloon
-                .lock()
-                .unwrap()
-                .resize(balloon_size)
-                .map_err(Error::VirtioBalloonResizeFail)?;
-        }
-
-        Ok(balloon_size)
-    }
-
-    pub fn get_balloon_actual(&self) -> u64 {
-        if let Some(balloon) = &self.balloon {
-            return balloon.lock().unwrap().get_actual();
-        }
-
-        0
-    }
-
```

Comment:
So I think we're moving 20 lines of code from the memory manager to the device manager for simplifying the `--memory` CLI. My opinion here:

* The balloning is actually a memory resizing operation that I think makes sense to keep bound to the `--memory` API
* We're simplifying the `--memory` part of the CLI at the expense of making the overall CLI more complex by adding a `--balloon` device.

So imho it's not worth it.

Reply:
Thanks for your feedback @sameo 

>The balloning is actually a memory resizing operation that I think makes sense to keep bound to the --memory API

The `virtio-balloon` driver is responsible for hiding/showing some of the memory to the guest, but from a VMM perspective, this is nothing more than a virtio device that has nothing to do with how memory is mapped on the host. That's why I thought it could be separated.
And when adding more features in the future to `virtio-balloon`, the `--memory` parameter is gonna grow even bigger.

>We're simplifying the --memory part of the CLI at the expense of making the overall CLI more complex by adding a --balloon device.

I disagree on this one because I think it's easier to understand as a user that we're adding a `virtio-balloon` device by having a dedicated parameter for it, but that's only a different point of view.

>So imho it's not worth it.

Let's wait for @rbradford to give his opinion and I'll close the PR if he shares your opinion.

Reply:
> Thanks for your feedback @sameo
> 
> > The balloning is actually a memory resizing operation that I think makes sense to keep bound to the --memory API
> 
> The `virtio-balloon` driver is responsible for hiding/showing some of the memory to the guest, but from a VMM perspective, this is nothing more than a virtio device that has nothing to do with how memory is mapped on the host.
>

It does influence how memory is managed on the host or in the guest, and that's the whole point of it. Ballooning is purely a memory management parameter, and is perceived as a "soft/lazy" memory hotplug method. The fact that it happens to be based on virtio and how the VMM decides to hint the host based on what it receives on the virtio queues is an internal VMM implementation. Our UX show expose the memory balloning feature, not how things are internally implemented.

> That's why I thought it could be separated.

Having a `--ballon` parameter is not too bad per se, it's only exposing a feature and that's fine. I think it's not better than the current API because it decouple ballooning from the memory config while they're actually not.

> And when adding more features in the future to `virtio-balloon`, the `--memory` parameter is gonna grow even bigger.
> 
> > We're simplifying the --memory part of the CLI at the expense of making the overall CLI more complex by adding a --balloon device.
> 
> I disagree on this one because I think it's easier to understand as a user that we're adding a `virtio-balloon` device by having a dedicated parameter for it, but that's only a different point of view.
> 

The user should only know about ballooning, not the fact that there's a virtio device handling that feature as our internal implementation. And that's still hidden with `--balloon` btw :-)

Reply:
I support the idea of moving to `--balloon`:

* We already have some memory management controls outside of `--memory`, e.g. `--memory-zone`
* With more options being added to the balloon (see #1852) it make sense to not overload the `--memory` parameter
* It also provides some scope for simplifying `MemoryManager`
* If we do choose to remove the functionality in the future it makes it marginally easier.